Add Groth16 settlement pallet docs (#37)
HorizenLabs/zkverify-docs: docs/overview/03-proof_submission_interface/02-proof_submitter_flow.md

1. A proof submitter (rollup / zkApp) submits the proof via the [`submitProof`](../02-mainchain/05-mainchain_api.md#submitprooffflonk) extrinsic of the appropriate verification pallet.

>     Currently we use four types of verifier (fflonk, zksync-era, risc0, and Groth16 verifier). The proof leaf `value` will be:

Reviewer: ```suggestion
    Currently we have implemented four verifiers (fflonk, zksync-era, risc0, and Groth16 verifier). The proof leaf `value` will be:
```

    - fflonk:
        ```
        keccak256(“fflonk-”, keccak256(ScaleEncode(vk)), "-", public_inputs)
        ```
    - zksync-era:
        ```
        keccak256("zksync-", public_inputs)
        ```
    - risc0:
        ```
        keccak256("risc0-", vk, "-", public_inputs)
        ```
    - Groth16:
        ```
        keccak256(keccak256("groth16"), keccak256(ScaleEncode(vk)), keccak256(ScaleEncode(public_inputs)))
        ```

>     As it can be seen, at the moment the verifiers use different conventions for defining the proof leaf value, but work is currently ongoing to harmonize them.

Reviewer: ```suggestion
    At the moment, the verifiers use different conventions for defining the proof leaf value, but work is currently ongoing to unify them.
```
